[raisinbread] Can't open bmi instrument

**Describe the bug**
When you try to open the bmi instrument you get an error
Example DCCID: 587630 > V1 > bmi
![Image](https://github.com/user-attachments/assets/684f8c79-5214-46b3-9e17-f84525eb4792)

![Image](https://github.com/user-attachments/assets/76d88851-7fb9-4418-9a7f-3fb227fafba1)
`
[Fri Feb 21 13:36:33.651943 2025] [php:notice] [pid 9482] [client 192.168.122.1:39298] [ERROR] Invalid field Examiner for instrument bmi#0 /var/www/Loris/php/libraries/NDB_BVL_Instrument.class.inc(1018): NDB_BVL_Instrument->getFieldValue()\n#1 /var/www/Loris/php/libraries/NDB_BVL_Instrument.class.inc(900): NDB_BVL_Instrument->_getExaminerNames()\n#2 /var/www/Loris/php/libraries/NDB_BVL_Instrument_LINST.class.inc(450): NDB_BVL_Instrument->_addMetadataFields()\n#3 /var/www/Loris/php/libraries/NDB_BVL_Instrument.class.inc(297): Loris\\Behavioural\\NDB_BVL_Instrument_LINST->loadInstrumentFile()\n#4 /var/www/Loris/modules/instruments/php/module.class.inc(81): NDB_BVL_Instrument::factory()\n#5 /var/www/Loris/src/Middleware/ResponseGenerator.php(51): LORIS\\instruments\\Module->handle()\n#6 /var/www/Loris/src/Middleware/AuthMiddleware.php(64): LORIS\\Middleware\\ResponseGenerator->process()\n#7 /var/www/Loris/src/Router/ModuleRouter.php(75): LORIS\\Middleware\\AuthMiddleware->process()\n#8 /var/www/Loris/src/Middleware/ExceptionHandlingMiddleware.php(55): LORIS\\Router\\ModuleRouter->handle()\n#9 /var/www/Loris/src/Router/BaseRouter.php(138): LORIS\\Middleware\\ExceptionHandlingMiddleware->process()\n#10 /var/www/Loris/src/Middleware/ResponseGenerator.php(51): LORIS\\Router\\BaseRouter->handle()\n#11 /var/www/Loris/src/Middleware/ContentLength.php(53): LORIS\\Middleware\\ResponseGenerator->process()\n#12 /var/www/Loris/htdocs/index.php(74): LORIS\\Middleware\\ContentLength->process()\n#13 {main}, referer: https://test-dev-270.loris.ca/instrument_list/?candID=587630&sessionID=1053`

--- a/php/libraries/NDB_BVL_Instrument.class.inc
+++ b/php/libraries/NDB_BVL_Instrument.class.inc
@@ -854,10 +854,11 @@ abstract class NDB_BVL_Instrument extends NDB_Page
 
         $this->instanceData = $newData;
     }
-
     /**
      * Adds metadata fields (such as Examiner and Date_taken) to the
-     * current form
+     * current form. Note that NDB_BVL_Instrument_LINST.class.inc mimics the logic
+     * of this function to build the data dictionary, so any changes here should
+     * be reflected there as well.
      *
      * @return void
      * @access private

--- a/php/libraries/NDB_BVL_Instrument_LINST.class.inc
+++ b/php/libraries/NDB_BVL_Instrument_LINST.class.inc
@@ -447,6 +447,10 @@ class NDB_BVL_Instrument_LINST extends \NDB_BVL_Instrument
                             $pieces[1]
                         );
                     }
+                    // The sequence of if statements below building the data dictionary is designed to
+                    // mimic the same logic in the _addMetadataFields() function of
+                    // NDB_BVL_Instrument to keep the dictionary in sync with the fields added 
+                    // to the form, make sure to modify it accordingly
                     if ($this->DataEntryType!=="DirectEntry") {
                         $this->dictionary = array_merge(
                             $this->dictionary,

--- a/php/libraries/NDB_BVL_Instrument_LINST.class.inc
+++ b/php/libraries/NDB_BVL_Instrument_LINST.class.inc
@@ -440,16 +440,6 @@ class NDB_BVL_Instrument_LINST extends \NDB_BVL_Instrument
                     break;
                 case 'title':
                     $this->fullName = trim($pieces[1]);
-                    if ($addElements) {
-                        $this->form->addElement(
-                            'header',
-                            'instrument_title',
-                            $pieces[1]
-                        );
-                        if ($this->DataEntryType!=="DirectEntry") {
-                            $this->_addMetadataFields();
-                        }
-                    }
                     $this->dictionary = array_merge(
                         $this->dictionary,
                         [
@@ -523,6 +513,16 @@ class NDB_BVL_Instrument_LINST extends \NDB_BVL_Instrument
                             ),
                         ]
                     );
+                    if ($addElements) {
+                        $this->form->addElement(
+                            'header',
+                            'instrument_title',
+                            $pieces[1]
+                        );
+                        if ($this->DataEntryType!=="DirectEntry") {
+                            $this->_addMetadataFields();
+                        }
+                    }
                     break;
                 case 'begingroup':
                     if ($addElements) {

--- a/php/libraries/NDB_BVL_Instrument_LINST.class.inc
+++ b/php/libraries/NDB_BVL_Instrument_LINST.class.inc
@@ -447,7 +447,7 @@ class NDB_BVL_Instrument_LINST extends \NDB_BVL_Instrument
                             $pieces[1]
                         );
                     }
-                    // The sequence of if statements below building the data dictionary
+                    // The sequence of if statements below building the data dict
                     // mimics the same logic in the _addMetadataFields() function of
                     // NDB_BVL_Instrument to keep the dictionary in sync with the
                     // fields added to the form, make sure to modify it accordingly

--- a/php/libraries/NDB_BVL_Instrument_LINST.class.inc
+++ b/php/libraries/NDB_BVL_Instrument_LINST.class.inc
@@ -447,10 +447,10 @@ class NDB_BVL_Instrument_LINST extends \NDB_BVL_Instrument
                             $pieces[1]
                         );
                     }
-                    // The sequence of if statements below building the data dictionary is designed to
-                    // mimic the same logic in the _addMetadataFields() function of
-                    // NDB_BVL_Instrument to keep the dictionary in sync with the fields added 
-                    // to the form, make sure to modify it accordingly
+                    // The sequence of if statements below building the data dictionary
+                    // mimics the same logic in the _addMetadataFields() function of
+                    // NDB_BVL_Instrument to keep the dictionary in sync with the
+                    // fields added to the form, make sure to modify it accordingly
                     if ($this->DataEntryType!=="DirectEntry") {
                         $this->dictionary = array_merge(
                             $this->dictionary,

--- a/php/libraries/NDB_BVL_Instrument_LINST.class.inc
+++ b/php/libraries/NDB_BVL_Instrument_LINST.class.inc
@@ -440,45 +440,67 @@ class NDB_BVL_Instrument_LINST extends \NDB_BVL_Instrument
                     break;
                 case 'title':
                     $this->fullName = trim($pieces[1]);
-                    $this->dictionary = array_merge(
-                        $this->dictionary,
-                        [
-                            new DictionaryItem(
-                                $this->testName.'_Date_taken',
-                                'Date of Administration',
-                                $scope,
-                                new \LORIS\Data\Types\DateType(),
-                                new Cardinality(Cardinality::SINGLE),
-                                'Date_taken',
-                            ),
-                        ]
-                    );
-                    if (strrpos($this->testName ?? '', '_proband') === false) {
-                        if (!$this->postMortem) {
+                    if ($addElements) {
+                        $this->form->addElement(
+                            'header',
+                            'instrument_title',
+                            $pieces[1]
+                        );
+                    }
+                    if ($this->DataEntryType!=="DirectEntry") {
+                        $this->dictionary = array_merge(
+                            $this->dictionary,
+                            [
+                                new DictionaryItem(
+                                    $this->testName.'_Date_taken',
+                                    'Date of Administration',
+                                    $scope,
+                                    new \LORIS\Data\Types\DateType(),
+                                    new Cardinality(Cardinality::SINGLE),
+                                    'Date_taken',
+                                ),
+                            ]
+                        );
+                        if (strrpos($this->testName ?? '', '_proband') === false) {
+                            if (!$this->postMortem) {
+                                $this->dictionary = array_merge(
+                                    $this->dictionary,
+                                    [
+                                        new DictionaryItem(
+                                            $this->testName.'_Candidate_Age',
+                                            'Candidate Age (Months)',
+                                            $scope,
+                                            new \LORIS\Data\Types\Duration(),
+                                            new Cardinality(Cardinality::SINGLE),
+                                            'Candidate_Age',
+                                        ),
+                                    ]
+                                );
+                            } else {
+                                $this->dictionary = array_merge(
+                                    $this->dictionary,
+                                    [
+                                        new DictionaryItem(
+                                            $this->testName.'_Candidate_Age',
+                                            'Candidate Age at Death (Months)',
+                                            $scope,
+                                            new \LORIS\Data\Types\Duration(),
+                                            new Cardinality(Cardinality::SINGLE),
+                                            'Candidate_Age',
+                                        ),
+                                    ]
+                                );
+                            }
                             $this->dictionary = array_merge(
                                 $this->dictionary,
                                 [
                                     new DictionaryItem(
-                                        $this->testName.'_Candidate_Age',
-                                        'Candidate Age (Months)',
+                                        $this->testName.'_Window_Difference',
+                                        'Window difference from test battery (days)',
                                         $scope,
                                         new \LORIS\Data\Types\Duration(),
                                         new Cardinality(Cardinality::SINGLE),
-                                        'Candidate_Age',
-                                    ),
-                                ]
-                            );
-                        } else {
-                            $this->dictionary = array_merge(
-                                $this->dictionary,
-                                [
-                                    new DictionaryItem(
-                                        $this->testName.'_Candidate_Age',
-                                        'Candidate Age at Death (Months)',
-                                        $scope,
-                                        new \LORIS\Data\Types\Duration(),
-                                        new Cardinality(Cardinality::SINGLE),
-                                        'Candidate_Age',
+                                        'Window_Difference',
                                     ),
                                 ]
                             );
@@ -487,39 +509,19 @@ class NDB_BVL_Instrument_LINST extends \NDB_BVL_Instrument
                             $this->dictionary,
                             [
                                 new DictionaryItem(
-                                    $this->testName.'_Window_Difference',
-                                    'Window difference from test battery (days)',
+                                    $this->testName.'_Examiner',
+                                    'Examiner',
                                     $scope,
-                                    new \LORIS\Data\Types\Duration(),
+                                    // This should be an enum of examiners, but
+                                    // getExaminerNames currently returns an empty
+                                    // array if CommentID is not set.
+                                    new StringType(255),
                                     new Cardinality(Cardinality::SINGLE),
-                                    'Window_Difference',
+                                    'Examiner',
                                 ),
                             ]
                         );
-                    }
-                    $this->dictionary = array_merge(
-                        $this->dictionary,
-                        [
-                            new DictionaryItem(
-                                $this->testName.'_Examiner',
-                                'Examiner',
-                                $scope,
-                                // This should be an enum of examiners, but
-                                // getExaminerNames currently returns an empty
-                                // array if CommentID is not set.
-                                new StringType(255),
-                                new Cardinality(Cardinality::SINGLE),
-                                'Examiner',
-                            ),
-                        ]
-                    );
-                    if ($addElements) {
-                        $this->form->addElement(
-                            'header',
-                            'instrument_title',
-                            $pieces[1]
-                        );
-                        if ($this->DataEntryType!=="DirectEntry") {
+                        if ($addElements) {
                             $this->_addMetadataFields();
                         }
                     }

